bmc/core/filter_functions.h

```diff
@@ -0,0 +1,221 @@
+/**
+ * DSVerifier - Digital Systems Verifier
+ *
+ * Federal University of Amazonas - UFAM
+ *
+ * Authors:       Daniel Mello <dani-dmello@hotmail.com>
+ *                
+ * ------------------------------------------------------------------------
+ *
+ * Common functions needed for digital filters and frequency domain operations
+ *
+ * ------------------------------------------------------------------------
+ */
+
+
+/**
+ * Sine aproximation by taylor series (max 6 terms). 
+ * Algorithm created aiming to avoid state explosion due to loops and recursion.  
+ */
+ 
+double sinTyl(double x, int precision){
+    double sine;
+    double xsquared = x*x;
+    double aux;
+
+    if (precision >= 0){
+        aux = 0;
+        sine = aux;
+
+         if (precision >= 1){
+            aux = x; 
+            sine += aux; 
+
+            if (precision >= 2){
+                aux = aux*xsquared; 
+                sine -= aux/6;
+            
+                if (precision >= 3){
+                    aux = aux*xsquared;
+                    sine +=aux/120;
+                
+                    if(precision >=4){
+                        aux = aux*xsquared;
+                        sine -=aux/5040;
+
+                        if(precision >= 5){
+                            aux = aux*xsquared;
+                            sine +=aux/362880;
+
+                            if(precision >= 6){
+                                aux = aux*xsquared;
+                                sine -=aux/39916800;
+                            }
+                        }                     
+                    }
+                }
+            }
+        }
+    }
+    return (double)sine;
+}
+
+/**
+ * Cosine aproximation by taylor series (max 6 terms). 
+ * Algorithm created aiming to avoid state explosion due to loops and recursion.  
+ */
+double cosTyl(double x, int precision){
+    double cosine;
+    double xsquared = x*x;
+    double aux;
+
+    if (precision >= 0){
+        aux = 0;
+        cosine = aux;
+
+         if (precision >= 1){
+            aux = 1; 
+            cosine = 1; 
+
+            if (precision >= 2){
+                aux = xsquared; 
+                cosine -= aux/2; 
+            
+                if (precision >= 3){
+                    aux = aux*xsquared; 
+                    cosine += aux/24;
+                
+                    if(precision >=4){
+                        aux = aux*xsquared;
+                        cosine -=aux/720;
+
+                        if(precision >= 5){
+                            aux = aux*xsquared;
+                            cosine +=aux/40320;
+
+                            if(precision >= 6){
+                                aux = aux*xsquared;
+                                cosine -=aux/3628800;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Can you please add an assertion for checking whether the user has provided a precision that is not supported?